From 84e0d0341595a210a39072ef5f1447f6193cf243 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Thu, 14 Jan 2021 00:28:59 +0100
Subject: [PATCH 3/3] ARM: imx6: dh-imx6: Add update_sf script to install
 U-Boot into SF

Add script to read U-Boot from SD card and write it to matching
locations in the SPI NOR, thus making the SPI NOR bootable.

Upstream-Status: Inappropriate [custom environment]
Signed-off-by: Marek Vasut <marex@denx.de>
---
 include/configs/dh_imx6.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/dh_imx6.h b/include/configs/dh_imx6.h
index 008a70a7c2e..6c7c7171a4a 100644
--- a/include/configs/dh_imx6.h
+++ b/include/configs/dh_imx6.h
@@ -102,6 +102,10 @@
 	"ramdisk_addr_r=0x18000000\0"	\
 	"scriptaddr=0x14000000\0"	\
 	"fdtfile=imx6q-dhcom-pdk2.dtb\0"\
+	"update_sf=" /* Erase SPI NOR and install U-Boot from SD */	\
+		"load mmc 0:1 ${loadaddr} /boot/u-boot-with-spl.imx && "\
+		"sf probe && sf erase 0x0 0xf0000 && "			\
+		"sf write ${loadaddr} 0x400 ${filesize}\0"		\
 	BOOTENV
 
 #define CONFIG_BOOTCOMMAND		"run distro_bootcmd"
-- 
2.29.2

